### ISSUE TITLE
Using on Windows: `IndexTableReadingException`

### ISSUE THREAD

**Issue by Mukhammadsaid19** [hfst/hfst-ospell#56]
When I try to build `hfst-ospell` in Windows and use it with FST models, it throws `IndexTableReadingException` in function `void IndexTable::read(FILE *f, TransitionTableIndex number_of_table_entries)`:

![image](https://user-images.githubusercontent.com/56883018/168981230-377644c4-8a5d-4db7-bdd8-0f6ae93b5b57.png)

The same bug persists when using pre-compiled executable listed in the Apertium website:

![image](https://user-images.githubusercontent.com/56883018/168978210-e0090ef8-29b8-4d7b-8d6b-4ea6573f99cf.png)

Lexicon and error model files work well on Linux Ubuntu. What might be a problem?

**Comment by TinoDidriksen** [comment]
The native Windows binaries are not a priority because WSL works so perfectly, so nobody tests on native Windows.

**Comment by Mukhammadsaid19** [comment]
I would like to use it inside C# (for VSTO Microsoft Word Add-In) and I planned to bind it through DDL file. Initially I wanted to use `voikko` but it has many features that differ from the Uzbek language, so I decided to start from scratch. 

@snomos Interesting, I will check it out, thank you!
@TinoDidriksen Anyone used hfst-ospell and its dependents for MS Office Add-Ins?

**Comment by TinoDidriksen** [comment]
> Anyone used hfst-ospell and its dependents for MS Office Add-Ins?

Yes, I do that. Divvun also does that. But it's not the correct way any longer. VSTO extensions are headed to the scrap heap because they can't run on macOS, iPad, or web editions. Instead we have moved to Office.js add-ins that work cross-platform:
- https://docs.microsoft.com/office/dev/add-ins/
- https://github.com/GrammarSoft/proofing-gasmso
- https://github.com/divvun/divvun-gramcheck-web

What language are you trying to add a checker for?

**Comment by TinoDidriksen** [comment]
We are painfully aware that Office.js is limited (and I've [reported it](https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/proofing-tool-api-spelling-grammar-punctuation-hyphenation/idi-p/3052575) upstream, [twice](http://web.archive.org/web/20201021221732/https://officespdev.uservoice.com/forums/224641-feature-requests-and-feedback/suggestions/10475778-proofing-tool-api)), but it's still the only future-proof and cross-platform solution.

Divvun also makes keyboards. There's a whole pipeline for turning an FST into spellers, keyboards, and prediction, all for both desktop and mobile. @snomos can point you at docs. As for Uzbek, you may also be interested in https://github.com/apertium/apertium-uzb

Btw, we are on IRC on irc.oftc.net channels `#hfst` and `#apertium`

**Comment by snomos** [comment]
https://github.com/divvun/kbdgen2 takes a (relatively) simple yaml file (wrapped in a bundle with some metadata) as input, and produces keyboard packages for iOS, Android, Linux, Windows, macOS and ChromeOS. For iOS and Android, the keyboards can be bundled with Hfst-based spellers. `kbdgen2` is still work in progress. Other repos relevant to keyboards and spelling checkers are:

- https://github.com/divvun/divvun-keyboard - Android and iOS app for released keyboards 
- https://github.com/divvun/divvun-dev-keyboard - Android and iOS app for keyboards in development (and new keyboard features)
- https://github.com/giellalt, search for keyboard

**Comment by TinoDidriksen** [comment]
The nightly Windows builds got updated last week, so check if latest binary still throws.

**Comment by GooDeeJAY** [comment]
Was facing the same issue. In my case, opening a file in "rb" mode instead of "r", solved the problem.
